clients/config-center-client/src/main/java/org/apache/servicecomb/config/center/client/AddressManager.java

```diff
@@ -47,16 +54,29 @@ private String formatAddress(String address) {
   }
 
   public String address() {
-    synchronized (this) {
-      this.index++;
-      if (this.index >= addresses.size()) {
-        this.index = 0;
-      }
-      return addresses.get(index);
-    }
+    return endpointAddress.getAvailableZoneAddress();
   }
 
   public boolean sslEnabled() {
-    return address().startsWith("https://");
+    isSSLEnable = address().startsWith("https://");
+    return isSSLEnable;
+  }
+
+  public EndpointAddress getEndpointAddress() {
+    return endpointAddress;
+  }
+
+  public void setEndpointAddress(EndpointAddress endpointAddress) {
+    this.endpointAddress = endpointAddress;
+  }
+
+  @Subscribe
+  public void onRefreshEndpointEvent(RefreshEndpointEvent event) {
+    if (null == event || event.getName() != "CseConfigCenter") {
+      return;
+    }
+    endpointAddress.setAvailableZone(event.getSameZone());
+    endpointAddress.setAvailableRegion(event.getSameRegion());
+    endpointAddress.refreshCache();
```

Comment:
Encapsulation. Maybe calcute address groups can encapsulte in AddressManager. Events only send the server list. e.g. List<AddressInfo> where AddresInfo has address, zone, az information.

Reply:
done